packages/web/src/components/nav/desktop/AccountDetails.tsx

```diff
@@ -159,8 +160,23 @@ const SignedOutView = () => (
 )
 
 const GuestView = () => {
-  const [guestEmail] = useLocalStorage('guestEmail', '')
-
+  const [guestEmailFromLocalStorage] = useLocalStorage(GUEST_EMAIL, '')
+
+  const [guestEmail, setGuestEmail] = useState(guestEmailFromLocalStorage)
+
+  useEffect(() => {
+    // Listen for changes to the guest email in local storage
+    function listenForStorage() {
+      const storedGuestEmail = localStorage.getItem(GUEST_EMAIL)
+      if (storedGuestEmail) {
+        setGuestEmail(JSON.parse(storedGuestEmail))
+      }
+    }
+    window.addEventListener(GUEST_EMAIL, listenForStorage)
```

Comment:
needed because useLocalStorage doesn't seem to be able to listen on event changes by itself

Reply:
yeah what is the point of a 3rd party hook that doesn't handle this...

Reply:
might be worth making a hook for this in web/hooks

Reply:
yea i think other ppl are having the same issue https://github.com/streamich/react-use/issues/2607